### PR TITLE
test: add test about reference table

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

**Comment by kenhys** [comment]
@kou could you review this PR, please?

**Comment by kenhys** [comment]
@kou I've fixed  test case based on recursive=dependent. (removed duplicated tests and tweak outout order)

File: test/command/suite/io_flush/recursive/dependent/data_column/create.test [hunk 1]
```diff
@@ -0,0 +1,11 @@
+table_create Users TABLE_HASH_KEY ShortText
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/data_column/create.test]: What is tested here?

File: test/command/suite/io_flush/recursive/dependent/plugin_register.test [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
+
+plugin_register query_expanders/tsv
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/plugin_register.test]: Why do we need this test?

File: test/command/suite/io_flush/recursive/dependent/table/loop_reference.test [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+table_create Groups TABLE_HASH_KEY Users
+column_create Groups name COLUMN_SCALAR ShortText
+
+column_create Users name COLUMN_SCALAR ShortText
+column_create Users group COLUMN_SCALAR Groups
+
+load --table Users
+[
+{"_key": "User A", "name": "A", "group": "Group A"}
+]
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/table/loop_reference.test]: Do we need to load data?

File: test/command/suite/io_flush/recursive/dependent/data_column/create.test (hunk 1 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/data_column/create.test]: @kou 
It is intended to compare flushed result before/after creating table.
Is this `create.test` itself should be removed?

File: test/command/suite/io_flush/recursive/dependent/plugin_register.test (hunk 2 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/plugin_register.test]: @kou I want to check that `plugin_register` doesn't affect target objects ( before/after `io_flush`,  listed `flush[...]` are same)
Should be removed?

File: test/command/suite/io_flush/recursive/dependent/table/loop_reference.test (hunk 3 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/table/loop_reference.test]: Ah, no need to load.
I'll fix.

File: test/command/suite/io_flush/recursive/dependent/data_column/create.test (hunk 1 above)
Comment (kou) [reply test/command/suite/io_flush/recursive/dependent/data_column/create.test]: I don't know why we should compare flushed result before/after creating table.
So I think that this file is needless.

File: test/command/suite/io_flush/recursive/dependent/data_column/remove.test [hunk 4]
```diff
@@ -0,0 +1,12 @@
+table_create Users TABLE_HASH_KEY ShortText
+column_create Users name COLUMN_SCALAR ShortText
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/data_column/remove.test]: What is tested here?

File: test/command/suite/io_flush/recursive/dependent/plugin_register.test (hunk 2 above)
Comment (kou) [reply test/command/suite/io_flush/recursive/dependent/plugin_register.test]: `plugin_register` isn't related to `io_flush`.
So I think that this file is needless.

File: test/command/suite/io_flush/recursive/dependent/table/create.test [hunk 5]
```diff
@@ -0,0 +1,11 @@
+table_create Users TABLE_HASH_KEY ShortText
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/table/create.test]: What is tested here?

File: test/command/suite/io_flush/recursive/dependent/table/remove.test [hunk 6]
```diff
@@ -0,0 +1,11 @@
+table_create Users TABLE_HASH_KEY ShortText
+
+#@collect-query-log true
+io_flush --recursive dependent
+#@collect-query-log false
```
Comment (kou) [inline test/command/suite/io_flush/recursive/dependent/table/remove.test]: What is tested here?

File: test/command/suite/io_flush/recursive/dependent/data_column/remove.test (hunk 4 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/data_column/remove.test]: @kou I want to test before/after `column_remove`, how io_flush output is changed.

File: test/command/suite/io_flush/recursive/dependent/table/create.test (hunk 5 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/table/create.test]: @kou I want to test before/after `table_create`, how io_flush output is changed.

File: test/command/suite/io_flush/recursive/dependent/table/remove.test (hunk 6 above)
Comment (kenhys) [reply test/command/suite/io_flush/recursive/dependent/table/remove.test]: @kou I want to test before/after `table_remove`, how io_flush output is changed.

**Comment by kenhys** [comment]
@kou I've removed all redundant test case.

**Pull request merged**